core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts

```diff
@@ -0,0 +1,636 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for exploration editor page component.
+ */
+
+// TODO(#7222): Remove the following block of unnnecessary imports once
+// the code corresponding to the spec is upgraded to Angular 8.
+import { UpgradedServices } from 'services/UpgradedServices';
+// ^^^ This block is to be removed.
+
+require('pages/exploration-editor-page/exploration-editor-page.component.ts');
+
+describe('Exploration editor page component', function() {
+  var ctrl = null;
+  var $q = null;
+  var $rootScope = null;
+  var $scope = null;
+  var $timeout = null;
+  var $uibModal = null;
+  var AutosaveInfoModalsService = null;
+  var ChangeListService = null;
+  var ContextService = null;
+  var EditabilityService = null;
+  var ExplorationFeaturesBackendApiService = null;
+  var ExplorationFeaturesService = null;
+  var ExplorationRightsService = null;
+  var ExplorationTitleService = null;
+  var ExplorationWarningsService = null;
+  var GraphDataService = null;
+  var PageTitleService = null;
+  var RouterService = null;
+  var StateEditorService = null;
+  var StateTopAnswersStatsBackendApiService = null;
+  var SiteAnalyticsService = null;
+  var ThreadDataService = null;
+  var UserExplorationPermissionsService = null;
+
+  var explorationId = 'exp1';
+  var explorationData = {
+    exploration_is_linked_to_story: true,
+    states: {
+      Introduction: {
+        param_changes: [],
+        content: {
+          html: '',
+          audio_translations: {}
+        },
+        unresolved_answers: {},
+        interaction: {
+          customization_args: {},
+          answer_groups: [],
+          default_outcome: {
+            param_changes: [],
```

Comment:
The outcome dict has more keys, isn't it?

Reply:
I'm not sure, but I think we don't need to use all the keys since it's a mock.